File: 0658-Find K Closest Elements/0658-Find K Closest Elements.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
        int n = arr.size();
        int index = lower_bound(arr.begin(), arr.end(), x) - arr.begin();
        
        int left = index - 1;
        int right = index;
        while (k--) {
            if (left >= 0 && (right == n || x - arr[left] <= arr[right] - x)) {
                --left;
            }
            else {
                ++right;
            }
        }
        
        return vector<int>(arr.begin() + left + 1, arr.begin() + right);        
    }
};

File: 0601-0700/0658-Find K Closest Elements/0658-Find K Closest Elements.py
import bisect
class Solution:
    def findClosestElements(self, arr: List[int], k: int, x: int) -> List[int]:
        right = bisect.bisect_left(arr, x)
        left = right - 1
        for _ in range(k):
            if left >= 0 and (right == len(arr) or x - arr[left] <= arr[right] - x):
                left -= 1
            else:
                right += 1
                
        return arr[left + 1: right]
